packages/harmony/src/components/input/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,178 @@
+  const shouldShowMaxLengthText =
+    isFocused &&
+    maxLength &&
+    characterCount >= showMaxLengthThreshold * maxLength
+  // Turn the maxlength text to the warning color whenever we hit a certain threshold (default 90%)
+  const showMaxlengthWarningColor =
+    maxLength && characterCount >= maxLengthWarningThreshold * maxLength
+
+  // Styles for the root of the input
+  const inputRootStyle = {
+    [styles.default]: size === TextInputSize.DEFAULT,
+    [styles.small]: size === TextInputSize.SMALL,
+    [styles.warning]: warningProp,
+    [styles.error]: error,
+    [styles.focused]: isFocused,
+    [styles.disabled]: disabled,
+    [styles.required]: required
+  }
+
+  // Styles for the input element itself
+  const inputElStyle = {
+    [styles.default]: size === TextInputSize.DEFAULT,
+    [styles.small]: size === TextInputSize.SMALL,
+    [styles.disabled]: disabled
+  }
+
+  const inputRender = (
+    <Flex
+      className={cn(styles.inputAdornmentRow, { [styles.focused]: isFocused })}
+    >
+      {startAdornmentText && shouldShowAdornments ? (
+        <Text variant='label' size='l' color='subdued'>
+          {startAdornmentText}
+        </Text>
+      ) : null}
+      <input
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        ref={ref}
+        className={cn(styles.inputElement, inputClassName, inputElStyle)}
+        value={value}
+        maxLength={maxLength}
+        disabled={disabled}
+        placeholder={shouldShowPlaceholder ? placeholder : undefined}
+        aria-label={labelText ?? props['aria-label']}
+        aria-required={required}
+        id={id}
+        {...other}
+      />
+      {endAdornmentText && shouldShowAdornments ? (
+        <Text variant='label' size='l' color='subdued'>
+          {endAdornmentText}
+        </Text>
+      ) : null}
+    </Flex>
+  )
+
+  return (
+    <Flex className={cn(styles.root, className)} direction='column'>
+      <label
+        htmlFor={id}
+        className={cn(styles.inputRoot, inputRootClassName, inputRootStyle)}
+      >
+        {StartIcon ? <StartIcon /> : null}
+        <div className={styles.innerRootContainer}>
+          {shouldShowLabel ? (
+            <Flex
+              className={styles.labelRow}
+              direction='row'
+              alignItems='center'
+              justifyContent='space-between'
+              gap='s'
+            >
+              <Text
+                variant='body'
+                tag='span'
+                size='l'
+                className={cn(styles.label, {
+                  [styles.hasValue]: hasValue
+                })}
+              >
+                {labelText}
+              </Text>
+              {shouldShowMaxLengthText ? (
+                <Text
+                  variant='body'
+                  size='xs'
+                  tag='span'
+                  color={showMaxlengthWarningColor ? 'warning' : 'default'}
+                >
+                  {characterCount}/{maxLength}
```

Comment:
Do we want this as part of the label? (I'm thinking we probably do, but making sure!)

Reply:
Visually yes, we want it to stay aligned with the label. 
It won't actually be read out  to a screen reader as part of the label since we're providing label via `aria-label`
(from Figma)
![image](https://github.com/AudiusProject/audius-protocol/assets/6711655/9fe089ef-5d3a-4a0c-bab4-68a7f962f72f)

Reply:
Oh sorry, I missed that were always setting `aria-label`

Though I think if we went the non-wrapped label route maybe we can avoid `aria-label` in the non `hideLabel` case?